pkg/addon/controllers/cmainstallprogression/controller.go

```diff
@@ -200,3 +217,28 @@ func mergeInstallProgression(newobj, oldobj *addonv1alpha1.InstallProgression) {
 	}
 	newobj.Conditions = oldobj.Conditions
 }
+
+func overrideConfigMapByAddOnConfigs(
+	addOnConfigs []addonv1alpha1.AddOnConfig,
+	desiredConfigs map[addonv1alpha1.ConfigGroupResource][]addonv1alpha1.ConfigReferent,
+) {
+	gvkOverwritten := make(map[addonv1alpha1.ConfigGroupResource]struct{})
```

Comment:
can we uset Set generic here?

Reply:
yes, have modified the code to use sets.